codalab/rest/bundles.py

```diff
@@ -35,14 +35,14 @@
 )
 from codalab.rest.users import UserSchema
 from codalab.rest.util import get_bundle_infos, get_resource_ids, resolve_owner_in_keywords
-from codalab.server.authenticated_plugin import AuthenticatedPlugin
+from codalab.server.authenticated_plugin import AuthenticatedProtectedPlugin, ProtectedPlugin
 from codalab.worker.bundle_state import State
 from codalab.worker.download_util import BundleTarget
 
 logger = logging.getLogger(__name__)
 
 
-@get('/bundles/<uuid:re:%s>' % spec_util.UUID_STR)
+@get('/bundles/<uuid:re:%s>' % spec_util.UUID_STR, apply=ProtectedPlugin())
```

Comment:
What do you think if we remodel this around a "mixin"-type approach? This would mean that we could create the following classes that can then be combined (`apply` takes an array):

```
AuthenticatedPlugin # checks to see if users are authenticated (regardless of protected mode)
ProtectedPlugin # if protected mode is set, checks to see if users have been granted access
```

This means that our current classes would map to the following:

`apply=ProtectedPlugin()` -> `apply=ProtectedPlugin()`
`apply=AuthenticatedPlugin()` -> `apply=AuthenticatedPlugin()`
`apply=AuthenticatedProtectedPlugin()` -> `apply=[AuthenticatedPlugin(), ProtectedPlugin()]`

Really, this change would just involve removing `AuthenticatedProtectedPlugin()`, but I think it makes it easier to see what is going on and lets you compose more plugins in the future.

Reply:
I think that's a good idea too, but I don't foresee adding another plugin in the near future. Would it be okay if I keep it as is for now and maybe we can try this approach if we need to add more plugins?